extensions/common/aws/aws-s3-core/src/test/java/org/eclipse/edc/aws/s3/S3CoreExtensionTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (c) 2023 NTT DATA Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Masatake Iwasaki (NTT DATA) - Initial implementation
+ *
+ */
+
+package org.eclipse.edc.aws.s3;
+
+import org.eclipse.edc.junit.extensions.EdcExtension;
+import org.eclipse.edc.junit.testfixtures.MockVault;
+import org.eclipse.edc.spi.security.Vault;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.services.s3.model.GetUrlRequest;
+
+import java.util.Map;
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(EdcExtension.class)
```

Comment:
I think @jimmarino's point makes sense, this could be tested without spinning a whole edc runtime, there's another extension that can be used to unit-test extension classes (`DependencyInjectionExtension`)

Reply:
Removing the use of the integration test harness will also simplify the tests:

1. Vault can be a mock and verified
2. Contexts can be mocked

In general, it would be helpful to either have the test method names explicitly convey what is being tested or have comments added. For people quickly looking at the code, it's not apparent what the aim of the tests are.

Thanks